subsys/bluetooth/scan.c

```diff
@@ -597,22 +597,28 @@ static bool find_uuid(const u8_t *data,
 
 	for (size_t i = 0; i < data_len; i += uuid_len) {
 		struct bt_uuid *uuid;
-		u16_t uuid_16;
-		u32_t uuid_32;
+
+		struct bt_uuid_16 uuid_16 = {
+			.uuid.type = BT_UUID_TYPE_16,
+		};
+
+		struct bt_uuid_32 uuid_32 = {
+			.uuid.type = BT_UUID_TYPE_32,
+		};
+
 		struct bt_uuid_128 uuid_128 = {
 			.uuid.type = BT_UUID_TYPE_128,
 		};
 
-
 		switch (uuid_type) {
 		case BT_UUID_TYPE_16:
-			memcpy(&uuid_16, &data[i], uuid_len);
-			uuid = BT_UUID_DECLARE_16(sys_le16_to_cpu(uuid_16));
+			uuid_16.val = sys_get_le16(&data[i]);
+			uuid = (struct bt_uuid *)&uuid_16;
 			break;
 
 		case BT_UUID_TYPE_32:
-			memcpy(&uuid_32, &data[i], uuid_len);
-			uuid = BT_UUID_DECLARE_32(sys_le32_to_cpu(uuid_32));
+			uuid_32.val = sys_get_le32(&data[i]);
```

Comment:
Change is ok, we could set `uuid_len` to fixed value before since its used only for UUID_TYPE_128

Reply:
It is also used to increment loop counter because we can have a few UUID which have the same type in adv data field

Reply:
Ok, I missed it